Update assert_single description for clarity (#9093)
edgedb/edgedb: docs/reference/stdlib/set.rst

    .. index:: cardinality, singleton

>     Checks that the input set is empty or contains no more than one element.

Reviewer: Maybe even more clear:

```suggestion
    Checks that the input set has exactly zero or one elements.
```

wdyt?
